### PR TITLE
(Closes #1333) Unary minus parenthesizing

### LINKED ISSUE #1333: Unary operator output without brackets
The following invalid code is being output by the Fortran Writer

`d(j) = d(j) + -a(i) * y`

There should be brackets around the unary '-' operator.

The PSyIR looks like:

```
Assignment[]
    ArrayReference[name:'d']
        Reference[name:'j']
    BinaryOperation[operator:'ADD']
        ArrayReference[name:'d']
            Reference[name:'j']
        BinaryOperation[operator:'MUL']
            UnaryOperation[operator:'MINUS']
                ArrayReference[name:'a']
                    Reference[name:'i']
            Reference[name:'y']
```

### PR DESCRIPTION
**Pull request by JulienRemy**
Some very small changes to the Fortran backend so that `UnaryOperation.create(UnaryOperation.Operator.MINUS, ...)` nodes get parenthesized when needed.

Explicitly deals with power operations as `(-a) ** b` was being written as `-a ** b`, which is wrong due to operator precedence.

Also deals with the "grandparent" operator to avoid useless unary minus parenthesizing ie. when the parent binary operation is being parenthesized. 
Consider eg. `a + (-b) * c` and `a * (-b + c)`, both of which are correct Fortran (without compiler extensions) with minimal parentheses counts.

### CODE REVIEW

**Comment by arporter** [comment]
Hi @JulienRemy, thanks very much for this. Please could you fix the linting errors and then we'll be able to see how the test coverage looks.

**Comment by JulienRemy** [comment]
Hi @arporter, I just fixed flake8 complaints and codecov seems content enough. Should I request PR review from you on this one?

**Comment by arporter** [comment]
> Hi @arporter, I just fixed flake8 complaints and codecov seems content enough. Should I request PR review from you on this one?

Yes please, if you request from both myself and Sergi it might happen quicker :-)

**Comment by JulienRemy** [comment]
> > Hi @arporter, I just fixed flake8 complaints and codecov seems content enough. Should I request PR review from you on this one?
> 
> Yes please, if you request from both myself and Sergi it might happen quicker :-)

Okay, thanks!
@sergisiso I also requested review from you.

File: src/psyclone/tests/psyad/transformations/test_assignment_trans.py [hunk 1]
```diff
@@ -565,8 +566,6 @@ def test_single_valued_sub(tmpdir):
     check_adjoint(tl_fortran, active_variables, ad_fortran, tmpdir)
 
 
-@pytest.mark.xfail(reason="issue #1333 Unary '-' should be enclosed in "
```
Comment (arporter) [inline src/psyclone/tests/psyad/transformations/test_assignment_trans.py:568]: Great :-)

**Review by arporter — COMMENTED** [review]
Thanks very much for this Julien. All looks good, particular kudos for the comments. As this is a bug fix there's no need to update documentation.
I've grepped and there's no other mention of #1333 in the source, tests or docs.
At this stage I would run the integration tests but:

 * it doesn't seem possible to launch them for a branch in a forked repo
 * they're currently broken anyway (#2569)

**Comment by arporter** [comment]
Now that #2569 is on, the integration tests for compilation and NEMO should be working (but not LFRic) so I've created my own PR with this branch so that I can run them.

**Comment by arporter** [comment]
The NEMO and compilation integration tests passed so I can close this now.

**Review by arporter — APPROVED** [review]
All good now.
Will proceed to merge.

**Pull request merged**